sources/packages/backend/apps/queue-consumers/src/services/cas-invoice/cas-invoice.service.ts

```diff
@@ -0,0 +1,264 @@
+import { Injectable } from "@nestjs/common";
+import {
+  CASService,
+  PendingInvoicePayload,
+  InvoiceLineDetail,
+  SendPendingInvoicesResponse,
+} from "@sims/integrations/cas";
+import { CAS_BAD_REQUEST } from "@sims/integrations/constants";
+import {
+  CASInvoice,
+  CASInvoiceBatchApprovalStatus,
+  CASInvoiceStatus,
+  User,
+} from "@sims/sims-db";
+import {
+  processInParallel,
+  CustomNamedError,
+  getAbbreviatedDateOnlyFormat,
+} from "@sims/utilities";
+import { ProcessSummary } from "@sims/utilities/logger";
+import { DataSource, Repository } from "typeorm";
+import { InjectRepository } from "@nestjs/typeorm";
+import { SystemUsersService } from "@sims/services";
+
+@Injectable()
+export class CASInvoiceService {
+  constructor(
+    private readonly systemUsersService: SystemUsersService,
+    private readonly casService: CASService,
+    @InjectRepository(CASInvoice)
+    private readonly casInvoiceRepo: Repository<CASInvoice>,
+    private readonly dataSource: DataSource,
+  ) {}
+  /**
+   * Get the list of pending invoices to be sent from the approved batch.
+   * @returns list of pending invoices.
+   */
+  async getPendingInvoices(): Promise<CASInvoice[]> {
+    const pendingInvoices = await this.dataSource
+      .getRepository(CASInvoice)
+      .find({
+        select: {
+          id: true,
+          casSupplier: {
+            id: true,
+            supplierNumber: true,
+            supplierAddress: { supplierSiteCode: true },
+          },
+          createdAt: true,
+          invoiceNumber: true,
+          disbursementReceipt: {
+            id: true,
+            fileDate: true,
+            createdAt: true,
+          },
+          casInvoiceBatch: {
+            id: true,
+            batchName: true,
+            batchDate: true,
+            createdAt: true,
+          },
+          casInvoiceDetails: {
+            id: true,
+            valueAmount: true,
+            casDistributionAccount: {
+              id: true,
+              operationCode: true,
+              awardValueCode: true,
+              distributionAccount: true,
+            },
+          },
+        },
+        relations: {
+          casInvoiceBatch: true,
+          disbursementReceipt: true,
+          casSupplier: true,
+          casInvoiceDetails: {
+            casDistributionAccount: true,
+          },
+        },
+        where: {
+          casInvoiceBatch: {
+            approvalStatus: CASInvoiceBatchApprovalStatus.Approved,
+          },
+          invoiceStatus: CASInvoiceStatus.Pending,
+        },
+      });
+    return pendingInvoices;
+  }
+
+  /**
+   * Send pending invoices to CAS.
+   * @param parentProcessSummary parent process summary for logging.
+   * @param pendingInvoices pending invoices.
+   */
+
+  async sendInvoices(
```

Comment:
This method would reinforce the idea from here: https://github.com/bcgov/SIMS/pull/4377/files#r1974542325
Having it called from a consumer does not seem reasonable.